bats_test/upload-file/single-bundled-files.bats

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bats
+
+# Support lib
+load '/home/node/packages/node_modules/bats-support/load.bash'
+# Assertions
+load '/home/node/packages/node_modules/bats-assert/load.bash'
+
+setup_file() {
+    run $DIR/create-file.sh
+    echo "##### Created 10 chunk test file" >>/home/node/bats.log
+
+    # Start test from correct working directory
+    # TODO: Use a $HOME variable for local testing support
+    cd /home/node/ardrive-cli
+}
+
+@test "upload-file creates a bundled transaction by default" {
+    run -0 bash -c "yarn ardrive upload-file --dry-run -l '/home/node/10Chunks.txt' -F $PUB_FOLD_ID -w $WALLET | jq '.created[] .type'"
+
+    assert_line -n 0 '"file"'
+    assert_line -n 1 '"bundle"'
+    assert_line -n 2 ''
+
+}
+
+@test "upload-file creates v2 transactions when --no-bundle is specified" {
+    run -0 bash -c "yarn ardrive upload-file --no-bundle --dry-run -l '/home/node/10Chunks.txt' -F $PUB_FOLD_ID -w $WALLET | jq '.created[] .type'"
+
+    assert_line -n 0 '"file"'
+    assert_line -n 1 ''
+}
+
+@test "upload bundled file produces one fee" {
+    run -0 bash -c "yarn ardrive upload-file --dry-run -l '/home/node/10Chunks.txt' -F $PUB_FOLD_ID -w $WALLET | jq '.fees'"
+
+    assert_line -n 0 '{'
+    assert_line -n 1 --regexp '(\s){0,}("){1}(\w|-){43}("){1}'
```

Comment:
You could save yourself the some of the pageantry of the regex by using jq in raw mode (gets rid of the whitespace and quotes), getting the fee keys, and unpacking the output array , e.g.:
```
| jq -r '.fees | keys | .[]'
```
yielding output like:
```
7DPiA3gv0Hcsn8ILpYIhcOYvV0udjyUnMs7G3bDrLhI
```

Reply:
Added: 34ca7f3